File: tutorials/tour-of-restate-go/app/cartobject.go
package main

import restate "github.com/restatedev/sdk-go"

// <start_user_session>
type CartObject struct{}

func (CartObject) AddTicket(ctx restate.ObjectContext, ticketId string) (bool, error) {
	return true, nil
}

func (CartObject) Checkout(ctx restate.ObjectContext) (bool, error) {
	return true, nil
}

func (CartObject) ExpireTicket(ctx restate.ObjectContext, ticketId string) error {
	return nil
}

// <end_user_session>

File: typescript/tutorials/tour-of-restate-typescript/src/app/cart_object.ts
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

import * as restate from "@restatedev/restate-sdk";
import { TicketObject } from "./ticket_object";
import { CheckoutService } from "./checkout_service";

// <start_user_session>
export const cartObject = restate.object({
  name: "CartObject",
  handlers: {
    async addTicket(ctx: restate.ObjectContext, ticketId: string) {
      return true;
    },

    async checkout(ctx: restate.ObjectContext) {
      return true;
    },

    async expireTicket(ctx: restate.ObjectContext, ticketId: string) {},
  },
});
// <end_user_session>

export const CartObject: typeof cartObject = { name: "CartObject" };
